lib/exmagick.ex

```diff
@@ -191,52 +178,59 @@ defmodule ExMagick do
     init
   end
 
-  @spec init!() :: image
   def init! do
     {:ok, handle} = init()
     handle
   end
 
-  @spec init() :: {:ok, image} | {:error, reason :: term}
   @doc """
   Creates a new image handle with default values.
 
   Image attributes may be tuned by using the `attr/2` function.
   """
   def init, do: fail
 
-  @spec image_load!(image, Path.t) :: image
   def image_load!(handle, path) do
     {:ok, handle} = image_load(handle, path)
     handle
   end
 
-  @spec image_load(image, Path.t) :: {:ok, image} | {:error, reason :: term}
   @doc """
-  Loads into the handler an image from a file.
+  Loads an image into the handler. You may provide a file path or a
+  tuple `{:blob, ...}` which the second argument is the blob to load.
   """
-  def image_load(_handle, _path), do: fail
+  def image_load(handle, {:blob, blob}), do: image_load_blob(handle, blob)
+  def image_load(handle, path), do: image_load_file(handle, path)
 
-  @spec image_dump!(image, Path.t) :: image
   def image_dump!(handle, path) do
     {:ok, handle} = image_dump(handle, path)
     handle
   end
 
-  @spec image_dump(image, Path.t) :: {:ok, image} | {:error, reason :: term}
   @doc """
   Saves an image to one or multiple files.
 
   If the attr `:adjoin` is `false`, multiple files will be created and the
   filename is expected to have a printf-formatting sytle (ex.: `foo%0d.png`).
   """
-  def image_dump(_handle, _path), do: fail
+  def image_dump(handle, path), do: image_dump_file(handle, path)
 
-  @spec set_attr(image, atom, term) :: {:ok, image} | {:error, reason :: term}
-  defp set_attr(_handle, _attribute, _value), do: fail
+  @doc """
+  Returns the image as a binary. You can change the type of this image
+  using the `:magick` attribute.
+  """
+  def image_dump(handle), do: image_dump_blob(handle)
 
-  @spec get_attr(image, atom) :: {:ok, attr_value :: term} | {:error, reason :: term}
-  defp get_attr(_handle, _attribute), do: fail
+  def image_dump!(handle) do
+    {:ok, blob} = image_dump(handle)
+    blob
+  end
 
-  defp fail, do: {:error, "native function"}
+  defp image_load_file(_handle, _path), do: fail
+  defp image_load_blob(_handle, _blob), do: fail
+  defp image_dump_file(_handle, _path), do: fail
+  defp image_dump_blob(_handle), do: fail
+  defp set_attr(_handle, _attribute, _value), do: fail
+  defp get_attr(_handle, _attribute), do: fail
+  defp fail, do: raise "native function error"
```

Comment:
se der o fail é pq não carregou o NIF. podia mudar pra "ExMagick NIF not loaded"